cljfmt/src/cljfmt/tool.clj

```diff
@@ -62,66 +50,34 @@
             (assoc-in [:counts :error] 1)
             (assoc :exception ex))))))
 
-(defn- print-stack-trace [ex]
-  (when-not *no-output*
-    (binding [*out* *err*]
-      (st/print-stack-trace ex))))
-
-(defn- print-file-status [options status]
-  (let [path (project-path options (:file status))]
-    (when-let [ex (:exception status)]
-      (warn "Failed to format file:" path)
-      (print-stack-trace ex))
-    (when (:reformatted status)
-      (warn "Reformatted" path))
-    (when-let [diff (:diff status)]
-      (warn path "has incorrect formatting")
-      (println diff))))
-
 (defn- exit [counts]
   (when-not (zero? (:error counts 0))
     (System/exit 2))
   (when-not (zero? (:incorrect counts 0))
     (System/exit 1)))
 
-(defn- print-final-count [counts]
-  (let [error     (:error counts 0)
-        incorrect (:incorrect counts 0)]
-    (when-not (zero? error)
-      (warn error "file(s) could not be parsed for formatting"))
-    (when-not (zero? incorrect)
-      (warn incorrect "file(s) formatted incorrectly"))
-    (when (and (zero? incorrect) (zero? error))
-      (warn "All source files formatted correctly"))))
-
-(defn- merge-counts
-  ([]    zero-counts)
-  ([a]   a)
-  ([a b] (merge-with + a b)))
-
 (defn check-no-config
   "The same as `check`, but ignores dotfile configuration."
-  [options]
-  (let [map*   (if (:parallel? options) pmap map)
-        counts (->> (:paths options)
-                    (mapcat (partial find-files options))
-                    (map* (partial check-one options))
-                    (map (fn [status]
-                           (print-file-status options status)
-                           (:counts status)))
-                    (reduce merge-counts))]
-    (print-final-count counts)
-    (exit counts)))
+  [{:keys [parallel? paths report] :as options
+    :or   {report report/console}}]
+  (let [context (report :check/initial options nil)
+        map*    (if parallel? pmap map)
+        summary (->> paths
+                     (mapcat (partial find-files context))
+                     (map* (partial check-one context))
+                     (reduce #(report :check/file %1 %2) context))]
+    (report :check/summary summary nil)))
 
 (defn check
   "Checks that the Clojure paths specified by the :paths option are
   correctly formatted."
   [options]
-  (let [opts (merge (config/load-config) options)]
+  (let [opts (merge {:report report/console}
```

Comment:
Is the merge of `{:report report/console}` necessary given the `:or` set in `check-no-config`?

And similarly for `fix` and `fix-no-config`?

Reply:
Good catch. No they are not! 😁 Will fix shortly

Reply:
Fixed in https://github.com/weavejester/cljfmt/pull/342/commits/ddf28c0b753ebd0fd51015f800c59b3133ec0972